### PR TITLE
Add GetIPAddress()

### PR DESCRIPTION
**Pull request by amery**
Add GetIPAddresses() and a dumb GetIPAddress()

### CODE REVIEW

**Comment by amery** [comment]
this is mostly a call for discussion. @karasz 
1) should we introduce a new dedicated package to work with addresses here and on davarza?
2) what's the best way to choose what address are we most likely to use?

**Comment by amery** [comment]
or maybe use an existing third-party dependency? does harshicorp have something already?

File: net.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+			var s []byte
+
+			switch v := addr.(type) {
+			case *net.IPAddr:
+				s = v.IP
+			case *net.IPNet:
+				s = v.IP
+			}
+
+			if ip, ok := netip.AddrFromSlice(s); ok {
```
Comment (karasz) [inline net.go:82]: Well, if I want to be nitpicking (which I don't) I would say that there are 3 possible results that come from AddrFromSlice (IPv4 if the slice length is 4) and either a IPv6 mapped IPv4 or a standard IPv6 if the slice length is 16. The thing is that you cannot actually distinguish between the later two options easily, the question is do we care? at this point I think we do not care so I will approve.

**Review by karasz — APPROVED** [review]
LGTM comments on code.

File: net.go (hunk 1 above)
Comment (amery) [reply net.go:82]: I think those are supposed to be treated as IPv6

**Pull request merged**